docs/core/faq/branching.md

```diff
@@ -0,0 +1,174 @@
+
+# How do I make branches in my flow?
+
+Since Prefect flows are represented as DAGs it is easy to add branching logic to a flow. By default branches can be specified by setting upstreams and downstreams to tasks in the flow. To take it a step further, Prefect also has concepts built in to allow for conditional branching based on task outputs.
+
+### Adding branches to a flow
+
+Let's use the following example. I have a piece of data that I want to perform some arithmetic on and then output that resulting value. My flow would look something like this:
+
+![Flow with Branching](/faq/branching.png)
+
+By using Prefect's syntax definition for setting task upstreams and downstreams this can be accomplished without any extras.
+
+:::: tabs
+::: tab "Functional API"
+```python
+from prefect import task, Flow
+
+@task
+def get_number():
+    return 10
+
+@task
+def multiply(n):
+    return n * 10
+
+@task
+def divide(n):
+    return n / 10
+
+@task
+def output_value(value):
+    print(value)
+
+with Flow("branches") as flow:
+    n = get_number()
+
+    m = multiply(n)
+    d = divide(n)
+
+    output_value(m)
+    output_value(d)
+```
+:::
+
+::: tab "Imperative API"
+```python
+from prefect import Task, Flow
+
+class GetNumber(Task):
+    def run(self):
+        return 10
+
+class Multiply(Task):
+    def run(self, n):
+        return n * 10
+
+class Divide(Task):
+    def run(self, n):
+        return n / 10
+
+class OutputValue(Task):
+    def run(self, value):
+        print(value)
+
+flow = Flow("branches")
+
+n = GetNumber()
+
+m = Multiply()
+d = Divide()
+
+m.set_upstream(n, key="n", flow=flow)
+d.set_upstream(n, key="n", flow=flow)
+
+output_1 = OutputValue()
+output_2 = OutputValue()
+
+output_1.set_upstream(m, key="value", flow=flow)
+output_2.set_upstream(d, key="value", flow=flow)
+```
+:::
+::::
+
+### Branching based on conditional logic
+
+Prefect ships with various [tasks for control logic](/core/task_library/control_flow.html) that you can use off-the-shelf to control the branching structure of your flow. This is especially useful in cases where under certain conditions you want to control which branches of your flow run based on upstream events.
+
+Let's say you want to have a flow where some conditional task is evaluated and then the downstream branches are run depending on the output of that conditional task—true or false. An example of this might look like the following:
```

Comment:
```suggestion
Let's say you want to have a flow where some conditional task is evaluated and then the downstream branches are run depending on the output of that conditional task. An example of this might look like the following:
```